cosinnus_message/rocket_chat.py

```diff
@@ -66,8 +86,18 @@ def get_cached_rocket_connection(rocket_username, password, server_url, reset=Fa
             rocket_connection = None
 
     if rocket_connection is None:
-        rocket_connection = RocketChat(user=rocket_username, password=password, server_url=server_url, timeout=timeout)
-        cache.set(cache_key, rocket_connection, settings.COSINNUS_CHAT_CONNECTION_CACHE_TIMEOUT)
+        try:
+            session = requests.Session()
```

Comment:
This line means that we will create a new session for each RC connection and I'm not sure what the implications of this are. In production, we will have a single rocketchat cached connection for the backend admin "rocket-bot" and then another cached connection for **each** recently active user's RC user connection (for retrieving unread messages and such). With this implementation this means that we create a new Session for possibly thousands of connections, for each rocketchat-connection their own Session. 

Questions that arise from this: 

- is there a limit to the pooled sessions we can create? 
- should we maybe re-use the same Session object for all rocketchat-connections, since they will all connect only to the same domain of the assigned RC server in any case? this could increase our connection speed significantly. 
    - if so, do we need to handle when a Session becomes invalid, to create a new one?

Reply:
After some reading and testing I think we could use a single session for all rocket-chat connections. 
Pushed a suggested solution that we should discuss in more detail.